### PR TITLE
fix: change to allow use of websocket options

### PR DESCRIPTION
**Pull request by EdamAme-x**
Closes https://github.com/honojs/hono/issues/2997

### The author should do the following, if applicable

- [x] Add tests
- [x] Run tests
- [x] `bun run format:fix && bun run lint:fix` to format the code
- [x] Add [TSDoc](https://tsdoc.org/)/[JSDoc](https://jsdoc.app/about-getting-started) to document the code

### CODE REVIEW

**Comment by EdamAme-x** [comment]
typed
image

**Comment by yusukebe** [comment]
Hi @EdamAme-x 

Is this ready to review?

**Comment by EdamAme-x** [comment]
@yusukebe  yes. but
I currently have duplicate type definitions in deno.d.ts and websocket.ts, what should I do? I have tried several approaches, but the types will are broken. 
Or I think I can define the type with "any" handler options in deno.d.ts.

File: src/adapter/deno/deno.d.ts [hunk 1]
```diff
@@ -17,8 +17,16 @@ declare namespace Deno {
    */
   export function writeFile(path: string, data: Uint8Array): Promise<void>
 
-  export function upgradeWebSocket(req: Request): {
+  export function upgradeWebSocket(
+    req: Request,
+    options: UpgradeWebSocketOptions
```
Comment (EdamAme-x) [inline src/adapter/deno/deno.d.ts:22]:
```suggestion
    options: any
```

Comment (EdamAme-x) [reply src/adapter/deno/deno.d.ts:22]: UX is not degraded.

Comment (EdamAme-x) [reply src/adapter/deno/deno.d.ts]: I decided to use "any". Ready for review.

**Comment by EdamAme-x** [comment]
@yusukebe Ready for review

File: src/adapter/deno/deno.d.ts [hunk 2]
```diff
@@ -17,7 +17,10 @@ declare namespace Deno {
    */
   export function writeFile(path: string, data: Uint8Array): Promise<void>
 
-  export function upgradeWebSocket(req: Request): {
+  export function upgradeWebSocket(
+    req: Request,
+    options: any
```
Comment (yusukebe) [inline src/adapter/deno/deno.d.ts:22]: Add ` // eslint-disable-next-line @typescript-eslint/no-explicit-any` above it to avoid eslint error.

**Comment by yusukebe** [comment]
Hi @EdamAme-x 

I think it's not bad to use `any` in this case. I've left a comment; please check it!

**Comment by EdamAme-x** [comment]
thanks @yusukebe

**Comment by EdamAme-x** [comment]
@yusukebe ready

**Comment by EdamAme-x** [comment]
hi @yusukebe 
hmm... Github actions throw error

```console
info - 2024-06-21 06:36:46,045 -- ci service found: github-actions
Error: Codecov token not found. Please provide Codecov token with -t flag.
Error: Codecov:
                        Failed to properly create report: The process '/home/runner/work/_actions/codecov/codecov-action/v4/dist/codecov' 
```

**Comment by EdamAme-x** [comment]
However, the other tests were successful and can be merged.
@yusukebe 
What do you think?

**Comment by yusukebe** [comment]
@EdamAme-x 

Could you try to merge the current `main` into the branch for the PR and push it?

**Comment by EdamAme-x** [comment]
oh no... I made a typo.
But no problem.
The content of the actions does not seem to be updatable...
I'm ready. @yusukebe

**Comment by EdamAme-x** [comment]
I'll create new PR again.

**Comment by EdamAme-x** [comment]
hi @yusukebe 
I can't seem to get the github actions up to date.
But the test is successful, and the earlier codecov actions are also successful. I think this can be merged since the only thing I changed was the comment.

File: .github/workflows/ci.yml
Comment (yusukebe) [inline .github/workflows/ci.yml:1]: Can you remove this change which is unnecessary?

Comment (EdamAme-x) [reply .github/workflows/ci.yml:1]: removed

**Comment by EdamAme-x** [comment]
@yusukebe ready

**Review by yusukebe — APPROVED** [review]
LGTM!

**Comment by yusukebe** [comment]
@EdamAme-x Thank you very much!

Merging now.

**Pull request merged**